algorithms/ar-expsq/java/ExponentiationBySquaring.java

```diff
@@ -0,0 +1,31 @@
+class ExponentiationBySquaring {
+    static long N = 1000000007L; // prime modulo value
+    public static void main(String[] args)
+    {
+        long base = 5;
+        long exp = 100000;
+
+        long modulo = exponentiation(base, exp);
+        System.out.println(modulo);
+    }
+
+    static long exponentiation(long base, long exp)
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6356801)